Add Fluentd > cloudwatch log forwarder

kenden opened the pull request:
Picked up/Updated fluentd-cloudwatch changes from PR https://github.com/kubernetes/charts/pull/211

Now using image from https://github.com/fluent/fluentd-kubernetes-daemonset

The AWS keys are added as secrets from files, using Files.Glob (from PR
comment https://github.com/kubernetes/charts/pull/211#issuecomment-286662756

icereval committed:
fluentd cloudwatch incubator

icereval committed:
rename deployment -> daemonset

icereval committed:
annotations as value

icereval committed:
reorder values

icereval committed:
use helm namespace

icereval committed:
address code review

icereval committed:
move td-agent.conf into values.yaml

kenden committed:
Updated fluentd-cloudwatch code from PR 211

The configuration is now inside the container (from PR comment:
https://github.com/kubernetes/charts/pull/211#issuecomment-286643280),
I then removed the configMap holding the config.

The AWS keys are added as secrets from files, using Files.Glob (from PR
comment
https://github.com/kubernetes/charts/pull/211#issuecomment-286662756)

so0k commented:
I've been using a private helm chart while testing out several logging targets (logzio,loggly,logentries,...) and found that I still needed the config files in a configmap 😢 

Even though the images are slim (each tag only has the minimum required to target it's logging target), config often needed to be customised further through config files.

seanknox commented:
@k8s-bot ok to test

kenden committed:
Add configMap for fluentd config

Copied the config from what's in
/fluentd/etc in image:
fluent/fluentd-kubernetes-daemonset:v0.12-cloudwatch

kenden commented:
@so0k I re-added a configMap. It has the same config than what is inside the docker image 
fluent/fluentd-kubernetes-daemonset:v0.12-cloudwatch

unguiculus commented:
ping @mgoodness

unguiculus commented on incubator/fluentd-cloudwatch/templates/configmap.yaml:
```diff
@@ -0,0 +1,11 @@
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
```
`app: {{ template "name" . }}`

kenden replied:
done

unguiculus commented on incubator/fluentd-cloudwatch/Chart.yaml:
```diff
@@ -0,0 +1,18 @@
+sources:
+- https://github.com/kubernetes/charts
+- https://github.com/fluent/fluentd-kubernetes-daemonset
+maintainers:
```
Use Github usernames.

kenden replied:
done

unguiculus commented on incubator/fluentd-cloudwatch/Chart.yaml:
```diff
@@ -0,0 +1,18 @@
+name: fluentd-cloudwatch
+version: 0.1.0
```
Add `appVersion`

unguiculus replied:
Is there an `icon` you could add?

kenden replied:
done and done

unguiculus reviewed (changes requested):
We've started namespacing templates. Please apply this to your PR as well. See #1785.

kenden committed:
Update to use namespaces

kenden committed:
Add appVersion

kenden committed:
Add fluentd icon

kenden committed:
Using github usernames instead of real names

kenden commented:
@unguiculus I updated the Chart as requested

unguiculus commented on incubator/fluentd-cloudwatch/templates/configmap.yaml:
```diff
@@ -0,0 +1,11 @@
+metadata:
+  name: {{ template "fluentd-cloudwatch.fullname" . }}
+  labels:
+    app: {{ template "fluentd-cloudwatch.fullname" . }}
```
`app: {{ template "fluentd-cloudwatch.name" . }}`

unguiculus reviewed (changes requested):
Sorry for the delay. Almost there.

unguiculus commented:
/ok-to-test

kenden committed:
Update label for configmap

kenden commented:
@unguiculus Updated

unguiculus commented:
/lgtm

unguiculus commented:
Sorry this took so long in the end.
